Fiori Scenario from Neo to CF (#2164)
SAP/jenkins-library: documentation/docs/scenarios/ui5-sap-cp/Readme.md

# Build and Deploy SAPUI5 or SAP Fiori Applications on SAP Cloud Platform with Jenkins

> Build an application based on SAPUI5 or SAP Fiori with Jenkins and deploy the build result into an SAP Cloud Platform account in the Cloud Foundry environment.

Reviewer: We can still deploy to neo ....

> Build an application based on SAPUI5 or SAP Fiori with Jenkins and deploy the build result into an SAP Cloud Platform account in the Cloud Foundry environment. Optionally, deploy into the NEO environment.

Reviewer 2: ```suggestion
Build an application based on SAPUI5 or SAP Fiori with Jenkins and deploy the build result into an SAP Cloud Platform account in the Cloud Foundry environment. Optionally, deploy the build result into the Neo environment.
```

Build an application based on SAPUI5 or SAP Fiori with Jenkins and deploy the build result into an SAP Cloud Platform account in the Cloud Foundry environment.

## Prerequisites

> * You have installed the Java Runtime Environment 8.

Reviewer: I personally think we should rely on docker from that personal point of view the cleanup here is OK. BUT: there was never a corresponding decision by product management / product owner (these roles are not that clearly defined in this project ...). So we should still be able to run on a plain Jenkins with all the tools installed locally. Hence we should still document that.

Author: the central documentation isn't updated yet. i removed most prerequisites and pointed to the piper guided tour.

| File Name | Description | Position |
|-----|-----|-----|

> | [`mta.yaml`](https://github.com/SAP/jenkins-library/blob/master/documentation/docs/scenarios/ui5-sap-cp/files/mta.yaml) | This file controls the behavior of the [MTA toolset](https://sap.github.io/cloud-mta-build-tool/). | Place the `mta.yaml` file in your application root folder and adjust the values in brackets with your data. |

Reviewer 2: ```suggestion
| [`mta.yaml`](https://github.com/SAP/jenkins-library/blob/master/documentation/docs/scenarios/ui5-sap-cp/files/mta.yaml) | This file controls the behavior of the [MTA toolset](https://sap.github.io/cloud-mta-build-tool/). | Place the `mta.yaml` file in your application root folder and replace the values in brackets with your data. |
```

## Context

This scenario combines various different steps to create a complete pipeline.

> In this scenario, we want to show how to build an application based on SAPUI5 or SAP Fiori by using the multi-target application (MTA) concept and how to deploy the build result into an SAP Cloud Platform account in the Cloud Foundry environment. This document comprises the [mtaBuild](../../../steps/mtaBuild/) and the [cloudFoundryDeploy](../../../steps/cloudFoundryDeploy/) steps.

Reviewer 2: ```suggestion
In this scenario, we want to show how to build an application based on SAPUI5 or SAP Fiori by using the multi-target application (MTA) concept, and how to deploy the build result into an SAP Cloud Platform account in the Cloud Foundry environment. This document comprises the [mtaBuild](../../../steps/mtaBuild/) and the [cloudFoundryDeploy](../../../steps/cloudFoundryDeploy/) steps.
```

fioriOnCloudPlatformPipeline script:this
```

[fioriOnCloudPlatformPipeline](https://sap.github.io/jenkins-library/steps/fioriOnCloudPlatformPipeline) is a so called _scenario step_ that wraps mainly the [MTA Build](https://sap.github.io/jenkins-library/steps/mtaBuild/) step and the [SAP Cloud Platform Deployment](https://sap.github.io/jenkins-library/steps/cloudFoundryDeploy/) step.

### Scenario Configuration

> This is a basic configuration example, which also resides in the sources of the project. Edit `.pipeline/config.yml` below the root directory. The configuration corresponds to the steps wrapped in `fioriOnCloudPlatformPipeline`. Refer to the [mtaBuild](https://sap.github.io/jenkins-library/steps/mtaBuild/) and [cloudFoundryDeploy](https://sap.github.io/jenkins-library/steps/cloudFoundryDeploy/) documentation for details.

Reviewer 2: ```suggestion
This is a basic configuration example, which also resides in the sources of the project. Edit `.pipeline/config.yml` below the root directory. The configuration corresponds to the steps wrapped in `fioriOnCloudPlatformPipeline`. For more information, see [mtaBuild](https://sap.github.io/jenkins-library/steps/mtaBuild/) and [cloudFoundryDeploy](https://sap.github.io/jenkins-library/steps/cloudFoundryDeploy/).
```

Reviewer 2: What does "Edit `.pipeline/config.yml` below the root directory" mean?

```yaml
steps:
  mtaBuild:
    platform: 'CF'
  cloudFoundryDeploy:
    cloudFoundry:
      apiEndpoint: 'your-cf-endpoint' # default: 'https://api.cf.eu10.hana.ondemand.com'
      org: 'your-org-id'
      space: 'your-space-id'
      credentialsId: 'CF-jenkins-credentials-id'
```

### Optional environment

> Optionally, [fioriOnCloudPlatformPipeline](https://sap.github.io/jenkins-library/steps/fioriOnCloudPlatformPipeline) allows to deploy into  a NEO environment. Refer to the [neoDeploy](https://sap.github.io/jenkins-library/steps/neoDeploy/) documentation for details.

Reviewer 2: ```suggestion
Optionally, you can use [fioriOnCloudPlatformPipeline](https://sap.github.io/jenkins-library/steps/fioriOnCloudPlatformPipeline) to deploy your build result deploy into the Neo environment. For more information, see [neoDeploy](https://sap.github.io/jenkins-library/steps/neoDeploy/).
```

```yaml
steps:
  mtaBuild:
    buildTarget: 'NEO'
    platform: 'neo'
    platform: 'CF'
  cloudFoundryDeploy:
    cloudFoundry:
      apiEndpoint: 'your-cf-endpoint' # default: 'https://api.cf.eu10.hana.ondemand.com'
      org: 'your-org-id'
      space: 'your-space-id'
      credentialsId: 'CF-jenkins-credentials-id'
```

### Optional environment

Optionally, [fioriOnCloudPlatformPipeline](https://sap.github.io/jenkins-library/steps/fioriOnCloudPlatformPipeline) allows to deploy into  a NEO environment. Refer to the [neoDeploy](https://sap.github.io/jenkins-library/steps/neoDeploy/) documentation for details.

```yaml
steps:
  mtaBuild:
    platform: 'NEO'
  neoDeploy:
    neo:
      credentialsId: 'NEO_DEPLOY'
      credentialsId: 'NEO-jenkins-credentials-id'
      account: 'your-account-id'
      host: 'hana.ondemand.com'
```

#### Configuration for the MTA Build

| Parameter        | Description    |
| -----------------|----------------|
| `buildTarget`    | The target platform to which the mtar can be deployed. Possible values are: `CF`, `NEO`, `XSA` |

> | `mtaJarLocation` | The location of the multi-target application archive builder jar file, including file name and extension. |

Reviewer: This, especially the `mtaJarLocation` is really outdated and needs to be cleaned up  ...

Author: the central documentation isn't updated yet. i removed the complete MTA configuration paragraph including these properties.
